packages/esm-patient-immunizations-app/src/hooks/useImmunizations.tsx

```diff
@@ -0,0 +1,21 @@
+import { fhirBaseUrl, openmrsFetch } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+import { type FHIRImmunizationBundle } from '../types/fhir-immunization-domain';
+import { mapFromFHIRImmunizationBundle } from '../immunizations/immunization-mapper';
+
+export function useImmunizations(patientUuid: string) {
+  const immunizationsUrl = `${fhirBaseUrl}/Immunization?patient=${patientUuid}`;
+
+  const { data, error, isLoading, isValidating } = useSWR<{ data: FHIRImmunizationBundle }, Error>(
+    immunizationsUrl,
+    openmrsFetch,
+  );
+  const existingImmunizations = data ? mapFromFHIRImmunizationBundle(data.data) : null;
```

Comment:
Is there a reason not to just make `mapFromFHIRImmunizationBundle` local to this file? Usually our data fetching hooks are paired with the mapping code directly.

Reply:
I'm fine with having it local to the hook but the idea was having all mappers in one place. If you were observant there are two mappers local to a file named "immunization-mapper" which gives a sense of separation of concerns and makes sharing utils easier.

Reply:
Yeah, I did see that. It's not a big deal either way.